addons/hr_timesheet/models/project.py

```diff
@@ -125,6 +126,20 @@ def _convert_project_uom_to_timesheet_encode_uom(self, time):
         uom_to = self.env.company.timesheet_encode_uom_id
         return round(uom_from._compute_quantity(time, uom_to, raise_if_failure=False), 2)
 
+    def _model_domain(self):
+        if self.env.context.get('my_team'):
+            user = self.env.user.id
+            domain = [('user_id.employee_id.parent_id.user_id', '=', user), ('user_id', '!=', user)]
+            if self.env.context.get('my_project'):
+                domain = expression.OR([
+                    domain,
+                    [('user_id', '=', user)]
+                ])
+            project_id = self.search(domain)
+            return [('parent_res_model', '=', 'project.project'), ('parent_res_id', 'in', project_id.ids)]
+        return super()._model_domain()
```

Comment:
What is the all point of this ? 

- The method name is badly chosen, I don't understand what it is supposed to do.
- We are playing with context key in a way that indicated a bidouille to get different behaviors from different context, which is 99.99% of the time synonym of code that is difficult to maintain.

In conclusion, it seems to be a bad idea. Could you please explain what you try to achieve ? 

Thanks :)

Reply:
@tivisse As per the specs need to achieve this thing: https://nimb.ws/CVOzpT add the following filters: My Projects, My Team on ratings:
- In my project filter need to add project ratings of current user
- In my team filter need to add the project rating of current user's team
- When applying both filters at that time result should be combination of the filter.
Thanks.